src/timeMachine/utils/insertFunction.ts

```diff
@@ -24,3 +24,110 @@ export const generateImport = (
 
   return importStatement
 }
+
+export const getFluxExample = func => {
+  const {name, fluxType} = func
+
+  // "fluxType": "(<-tables:stream[A], every:duration, ?groupColumns:[string], ?unit:duration) => stream[B] where A: Record, B: Record"
+
+  // get copy of fluxtype signature before arrow sign
+  const arrowSign = fluxType.lastIndexOf('=') // using lastIndexOf because of edge cases like array.filter
+  const fluxParam = fluxType.slice(0, arrowSign) // fluxParam = (<-tables:stream[A], every:duration, ?groupColumns:[string], ?unit:duration)
```

Comment:
Looking at `aggregate.count` I get this when i console.log the variables out in an object
```
{
  arrowSign: 116,
  fluxParam: '(?column:A, ?fill:B) => {reduce:(state:int, values…t:(values:[D]) => int, fill:B, compute:(state:C) '
}
```

This looks wrong already. We might need to rethink this logic.

Also, can you move this util file to `shared` - since this gets used in both Time Machine and Notebooks. I don't think we should keep this in `timeMachine`. It might get mistakened in the future as Time Machine specific code which this isn't.

Reply:
good idea. I moved the `getFluxExample` function into its own file in the shared folder. 

my thought process there is; we need to separate the actual parameters from the rest of `fluxType` content. It's guaranteed everything to the left of the arrow sign contains the function's parameters, and everything to the right of the arrow is irrelevant information we wont need in the parsing process. 

Initially, I was using `fluxType.indexOf('=')` to find the position of the arrow sign, but this approach didn't work for functions like `array.filter`. Some flux types have multiple arrow signs, making it difficult to know where the parameters start and stop. So I figured using `lastIndexOf` would be a viable solution for this first step. But this means, we need to add logic that can weed out or clean up `fluxParam` and make sure we parse out the correct parameters. 

Because each function's fluxType is unique in some way, it's been difficult to write code that takes these nuances into account. I found it easier to have different blocks of code that can act like "barriers" and by the time the parameters get to the end, only the required parameters remain. 
With that being said, I'm absolutely all ears if you have suggestions here. What other approach can we take in order to separate the parameters from the rest of the content in the `fluxType` value?

Reply:
good idea. I moved the `getFluxExample` function into its own file in the shared folder. 

my thought process there is; we need to separate the actual parameters from the rest of `fluxType` content. It's guaranteed everything to the left of the arrow sign contains the function's parameters, and everything to the right of the arrow is irrelevant information we wont need in the parsing process. 

Initially, I was using `fluxType.indexOf('=')` to find the position of the arrow sign, but this approach didn't work for functions like `array.filter`. Some flux types have multiple arrow signs, making it difficult to know where the parameters start and stop. So I figured using `lastIndexOf` would be a viable solution for this first step. But this means, we need to add logic that can weed out or clean up `fluxParam` and make sure we parse out the correct parameters. 

Because each function's fluxType is unique in some way, it's been difficult to write code that takes these nuances into account. I found it easier to have different blocks of code that can act like "barriers" and by the time the parameters get to the end, only the required parameters remain. 
With that being said, I'm absolutely all ears if you have suggestions here. What other approach can we take in order to separate the parameters from the rest of the content in the `fluxType` value?